"AuthConfig" unrecognized field 'identitytoken' due config loading

Hello, we got the issue, using latest version of "docker-java" while trying to load and parse DOCKER configuration file ("`$HOME/.docker/config.json`").

It doesn't accept the field '`identitytoken`'. I have noticed that you have '`registrytoken`' field, which have been used since [DOCKER API v1.22](https://docs.docker.com/engine/api/v1.22/) version for the same purpose. I checked documentation, it was renamed there into --> '`identytoken`' just in the next [DOCKER API v1.23](https://docs.docker.com/engine/api/v1.23/) version.

Here is some stacktrace for more detailed info:

```
Caused by: java.lang.IllegalArgumentException: Unrecognized field "identitytoken" (class com.github.dockerjava.api.model.AuthConfig), not marked as ignorable (6 known properties: "serveraddress", "username", "auth", "password", "email", "registrytoken"])
 at [Source: N/A; line: -1, column: -1] (through reference chain: java.util.LinkedHashMap["auths"]->java.util.LinkedHashMap["hostname"]->com.github.dockerjava.api.model.AuthConfig["identitytoken"])
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3589)
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:3530)
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:3518)
	at com.github.dockerjava.core.AuthConfigFile.loadConfig(AuthConfigFile.java:120)
 
```

Looks, that should be enough to just rename '`@JsonProperty("identitytoken")`'.

--- a/src/main/java/com/github/dockerjava/api/model/AuthConfig.java
+++ b/src/main/java/com/github/dockerjava/api/model/AuthConfig.java
@@ -42,6 +42,12 @@ public class AuthConfig implements Serializable {
     @JsonProperty("registrytoken")
     private String registrytoken;
 
+    /**
+     * @since {@link com.github.dockerjava.core.RemoteApiVersion#VERSION_1_23}
+     */
+    @JsonProperty("identitytoken")
+    private String identitytoken;
+
     public String getUsername() {
         return username;
     }
@@ -84,6 +90,20 @@ public class AuthConfig implements Serializable {
 
     public AuthConfig withAuth(String auth) {
         this.auth = auth;
+        return this;
+    }
+
+    /**
+     * @see #identitytoken
+     */
+    public String getIdentitytoken() {
+        return identitytoken;
+    }
+    /**
+     * @see #identitytoken
+     */
+    public AuthConfig withIdentityToken(String identitytoken) {
+        this.identitytoken = identitytoken;
         return this;
     }
 

--- a/src/test/java/com/github/dockerjava/api/model/AuthConfigTest.java
+++ b/src/test/java/com/github/dockerjava/api/model/AuthConfigTest.java
@@ -61,4 +61,21 @@ public class AuthConfigTest {
 
         assertThat(authConfig1, equalTo(authConfig));
     }
+
+    @Test
+    public void compatibleWithIdentitytoken() throws IOException {
+        final ObjectMapper mapper = new ObjectMapper();
+        final JavaType type = mapper.getTypeFactory().uncheckedSimpleType(AuthConfig.class);
+        final AuthConfig authConfig = testRoundTrip(RemoteApiVersion.VERSION_1_23,
+                "/other/AuthConfig/docs1.json",
+                type
+        );
+        String auth = "YWRtaW46";
+        String identitytoken = "1cba468e-8cbe-4c55-9098-2c2ed769e885";
+        assertThat(authConfig, notNullValue());
+        assertThat(authConfig.getAuth(), is(auth));
+        assertThat(authConfig.getIdentitytoken(), is(identitytoken));
+        final AuthConfig authConfig1 = new AuthConfig().withAuth(auth).withIdentityToken(identitytoken);
+        assertThat(authConfig1, equalTo(authConfig));
+    }
 }

--- a/src/test/resources/samples/1.23/other/AuthConfig/docs1.json
+++ b/src/test/resources/samples/1.23/other/AuthConfig/docs1.json
@@ -1,0 +1,4 @@
+{
+  "auth": "YWRtaW46",
+  "identitytoken": "1cba468e-8cbe-4c55-9098-2c2ed769e885"
+}

